src/variorum/Nvidia/power_features.c

```diff
@@ -0,0 +1,241 @@
+
+            if (verbose)
+            {
+                fprintf(writedest, "_GPU_POWER_LIMIT Host: %s, Socket:%d, Device ID: %d, Power limit: %0.3lf\n",
+                        hostname, device_index, d, value);
+            }
+            else
+            {
+                if (!init_output)
+                {
+                    fprintf(writedest, "_GPU_POWER_LIMIT | Host | Socket | Device | PowerLimit\n");
+                    init_output = 1;
+                }
+                fprintf(writedest, "_GPU_POWER_LIMIT | %s | %d | %d | %0.3lf\n",
+                        hostname, device_index, d, value);
+            }
+        }
+    }
+    /*!@todo: Seperate interface for default power limits? */
+}
+
+void dump_clocks_data(FILE *writedest, int verbose)
+{
+    unsigned int gpu_clock;
+    int m_num_package = 2;
+    int gpus_per_socket = m_total_unit_devices / m_num_package;
+    int device_index = 1;
+    int d;
+    char hostname[1024];
+    static int init_output = 0;
+
+    gethostname(hostname, 1024);
+
+    /* Iterate over all GPU device handles populated at init and print GPU clock */
+    for (device_index = 0; device_index < 2; device_index++)
+    {
+        for (d = device_index * gpus_per_socket; d < (device_index + 1) * gpus_per_socket; ++d)
+        {
+            nvmlDeviceGetClock(m_unit_devices_file_desc[d], NVML_CLOCK_SM, NVML_CLOCK_ID_CURRENT, &gpu_clock);
+
+            if (verbose)
+            {
+                fprintf(writedest, "_GPU_CLOCKS Host: %s, Socket:%d, Device ID: %d, GPU Clock: %d\n",
+                        hostname, device_index, d, gpu_clock);
+            }
+            else
+            {
+                if (!init_output)
+                {
+                    fprintf(writedest, "_GPU_CLOCKS | Host | Socket | Device | Clock\n");
+                    init_output = 1;
+                }
+                fprintf(writedest, "_GPU_CLOCKS | %s | %d | %d | %d\n",
+                        hostname, device_index, d, gpu_clock);
+            }
+        }
+    }
+}
+
+void dump_gpu_utilization(FILE *writedest, int verbose)
+{
+    nvmlUtilization_t util;
+    char hostname[1024];
+    int m_num_package = 2;
+    int gpus_per_socket = m_total_unit_devices / m_num_package;
+    int device_index = 1;
+    int d;
+    static int init_output = 0;
+
+    gethostname(hostname, 1024);
+
+    /* Iterate over all GPU device handles populated at init and print GPU SM and memory utilization */
+    for (device_index = 0; device_index < 2; device_index++)
+    {
+        for (d = device_index * gpus_per_socket; d < (device_index + 1) * gpus_per_socket; ++d)
+        {
+            nvmlDeviceGetUtilizationRates(m_unit_devices_file_desc[d], &util);
+
+            if (verbose)
+            {
+                fprintf(writedest, "_GPU_UTILIZATION Host: %s, Socket: %d, Device ID: %d, GPU Utilization (%): SM: %d, Memory: %d\n",
+                    hostname, device_index, d, util.gpu, util.memory);
```

Comment:
Indent `hostname, device_index, d, util.gpu, util.memory);` on line 227 so it's even with `writedest`.

Reply:
Fixed.